stm32-modules/heater-shaker/tests/test_m115.cpp

```diff
@@ -23,6 +23,36 @@ SCENARIO("GetSystemInfo (M115) response works", "[gcode][parse][m115]") {
                 REQUIRE_THAT(buffer, Catch::Matchers::EndsWith(suffix));
             }
         }
+        WHEN("filling response with an unset serial number (all 0xFF)") {
+            std::array<char, SYSTEM_WIDE_SERIAL_NUMBER_LENGTH> TEST_SN;
+            memset(TEST_SN._M_elems, 0xFF, TEST_SN.size());
```

Comment:
whoa whoa we definitely should not be poking STL internals. arrays don't have convenient ctors but what they do have is [fill](https://en.cppreference.com/w/cpp/container/array/fill) which should do what you want